### PR TITLE
Update runtime.md

### PR DESCRIPTION
**Pull request by christopherrae**
Updated with additional content WWSO Security on Amazon GuardDuty EKS Protection (EKS Runtime Protection + EKS Audit Log Protection) and complimentary add-on services for S3, RDS, Lambda, and Malware protection.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/security/docs/runtime.md [hunk 1]
```diff
@@ -32,7 +32,41 @@ AppArmor is similar to seccomp, only it restricts an container's capabilities in
...
+
+#### Enable Amazon GuardDuty Malware Protection 
+Malware Protection helps you detect the potential presence of malware by scanning the Amazon Elastic Block Store (Amazon EBS) volumes that are attached to the Amazon Elastic Compute Cloud (Amazon EC2) instances and container workloads. Malware Protection provides scan options where you can decide if you want to include or exclude specific Amazon EC2 instances and container workloads at the time of scanning. It also provides an option to retain the snapshots of Amazon EBS volumes attached to the Amazon EC2 instances or container workloads, in your GuardDuty accounts. The snapshots get retained only when malware is found and Malware Protection findings are generated. Malware Protection offers two types of scans to detect potentially malicious activity in your Amazon EC2 instances and container workloads – GuardDuty-initiated malware scan and On-demand malware scan. 
+
+For more information, visit [Amazon GuardDuty User Guide - Malware Protection in Amazon GuardDuty (https://docs.aws.amazon.com/guardduty/latest/ug/malware-protection.html)
+
+#### Enable Amazon GuardDuty S3 Protection
```
Comment (jicowan) [inline content/security/docs/runtime.md:54]: This is the EKS best practices guide. We are not including content for other AWS services. Please remove.

File: content/security/docs/runtime.md [hunk 2]
```diff
@@ -32,7 +32,41 @@ AppArmor is similar to seccomp, only it restricts an container's capabilities in
...
+S3 Protection helps Amazon GuardDuty monitor AWS CloudTrail data events for Amazon Simple Storage Service (Amazon S3) that include object-level API operations to identify potential security risks for data within your Amazon S3 buckets. GuardDuty monitors both AWS CloudTrail management events and AWS CloudTrail S3 data events to identify potential threats in your Amazon S3 resources. Both the data sources monitor different kinds of activities. Examples of CloudTrail management events for S3 include operations that list or configure Amazon S3 buckets, such as ListBuckets, DeleteBuckets, and PutBucketReplication. Examples of CloudTrail data events for S3 include object-level API operations, such as GetObject, ListObjects, DeleteObject, and PutObject.
+
+For more information, visit [Amazon GuardDuty User Guide - Amazon S3 Protection In Amazon GuardDuty](https://docs.aws.amazon.com/guardduty/latest/ug/s3-protection.html)
+
+#### Enable Amazon GuardDuty RDS Protection
```
Comment (jicowan) [inline content/security/docs/runtime.md:59]: Same as prior comment.

File: content/security/docs/runtime.md [hunk 3]
```diff
@@ -32,7 +32,41 @@ AppArmor is similar to seccomp, only it restricts an container's capabilities in
...
+RDS Protection in Amazon GuardDuty analyzes and profiles RDS login activity for potential access threats to your Amazon Aurora databases (Amazon Aurora MySQL-Compatible Edition and Aurora PostgreSQL-Compatible Edition). This feature allows you to identify potentially suspicious login behavior. RDS Protection doesn't require additional infrastructure; it is designed so as not to affect the performance of your database instances. When RDS Protection detects a potentially suspicious or anomalous login attempt that indicates a threat to your database, GuardDuty generates a new finding with details about the potentially compromised database. For more information visit [Amazon GuardDuty User Guide - GuardDuty RDS Protection](https://docs.aws.amazon.com/guardduty/latest/ug/rds-protection.html)
+
+For more information, visit [Amazon GuardDuty User Guide - Amazon GuardDuty RDS Protection](https://docs.aws.amazon.com/guardduty/latest/ug/malware-protection.html)
+
+#### Enable Amazon GuardDuty Lambda Protection
```
Comment (jicowan) [inline content/security/docs/runtime.md:64]: Same as prior comment.

**Comment by christopherrae** [comment]
Jeremy,

I thought from the comments in the email that I replied to the other day, such changes were welcome. (see screenshot below)

***@***.***


Regards,
Christopher

Christopher Rae
Principal Worldwide Security GTM Specialist
WWSO Security | Amazon Web Services

[signature_1667972620]<https://aws.amazon.com/security/>

Thoughts on our interaction? Provide feedback here.<https://feedback.aws.amazon.com/?ea=xrae&fn=Christopher&ln=Rae>

From: Jeremy Cowan ***@***.***>
Reply-To: aws/aws-eks-best-practices ***@***.***>
Date: Friday, September 1, 2023 at 5:21 AM
To: aws/aws-eks-best-practices ***@***.***>
Cc: "Rae, Christopher" ***@***.***>, Author ***@***.***>
Subject: Re: [aws/aws-eks-best-practices] Update runtime.md (PR #372)


@jicowan commented on this pull request.

________________________________

In content/security/docs/runtime.md<https://github.com/aws/aws-eks-best-practices/pull/372#discussion_r1312957822>:

> +#### Enable Amazon GuardDuty EKS Protection

+We strongly recommend using Amazon GuardDuty's one-click EKS Protection. EKS Runtime Monitoring provides runtime threat detection coverage for Amazon Elastic Kubernetes Service (Amazon EKS) nodes and containers within your AWS environment. EKS Runtime Monitoring uses a new GuardDuty security agent (EKS add-on) that adds runtime visibility into individual EKS workloads, for example, file access, process execution, and network connections. The GuardDuty security agent helps GuardDuty identify specific containers within your EKS clusters that are potentially compromised. It can also detect attempts to escalate privileges from an individual container to the underlying EC2 host, and the broader AWS environment.

+

+For more information, visit [Amazon GuardDuty User Guide - Configuring EKS Runtime Monitoring](https://docs.aws.amazon.com/guardduty/latest/ug/eks-protection-configuration.html)

+

+#### Enable Amazon GuardDuty EKS Audit Log Monitoring

+EKS Audit Log Monitoring helps you detect potentially suspicious activities in your EKS clusters within Amazon Elastic Kubernetes Service. When you enable EKS Audit Log Monitoring, GuardDuty immediately begins to monitor Kubernetes audit logs from your Amazon EKS clusters and analyze them for potentially malicious and suspicious activity. It consumes Kubernetes audit log events directly from the Amazon EKS control plane logging feature through an independent and duplicative stream of flow logs. This process does not require any additional set up or affect any existing Amazon EKS control plane logging configurations that you might have. When you disable EKS Audit Log Monitoring, GuardDuty immediately stops monitoring and analyzing the Kubernetes audit logs for your EKS resources.

+

+For more information, visit [Amazon GuardDuty User Guide - EKS Audit Log Monitoring](https://docs.aws.amazon.com/guardduty/latest/ug/guardduty-eks-audit-log-monitoring.html).

+

+#### Enable Amazon GuardDuty Malware Protection

+Malware Protection helps you detect the potential presence of malware by scanning the Amazon Elastic Block Store (Amazon EBS) volumes that are attached to the Amazon Elastic Compute Cloud (Amazon EC2) instances and container workloads. Malware Protection provides scan options where you can decide if you want to include or exclude specific Amazon EC2 instances and container workloads at the time of scanning. It also provides an option to retain the snapshots of Amazon EBS volumes attached to the Amazon EC2 instances or container workloads, in your GuardDuty accounts. The snapshots get retained only when malware is found and Malware Protection findings are generated. Malware Protection offers two types of scans to detect potentially malicious activity in your Amazon EC2 instances and container workloads – GuardDuty-initiated malware scan and On-demand malware scan.

+

+For more information, visit [Amazon GuardDuty User Guide - Malware Protection in Amazon GuardDuty (https://docs.aws.amazon.com/guardduty/latest/ug/malware-protection.html)

+

+#### Enable Amazon GuardDuty S3 Protection

This is the EKS best practices guide. We are not including content for other AWS services. Please remove.

**Comment by jicowan** [comment]
I already have someone working on the signing content so there's no need to add that now. The other content you've added for runtime monitoring, audit logs, malware, etc looks ok. I'll need to review it more thoroughly once you're finished with the edits.

**Review by jicowan — APPROVED** [review]

**Comment by christopherrae** [comment]
Awesome, thanks so much Jeremy!


Regards,
Christopher

From: Jeremy Cowan ***@***.***>
Reply-To: aws/aws-eks-best-practices ***@***.***>
Date: Friday, September 15, 2023 at 12:20 PM
To: aws/aws-eks-best-practices ***@***.***>
Cc: "Rae, Christopher" ***@***.***>, Author ***@***.***>
Subject: Re: [aws/aws-eks-best-practices] Update runtime.md (PR #372)


@jicowan approved this pull request.

**Pull request merged**